src/services/ResumeParserService.js

```diff
@@ -0,0 +1,269 @@
+/**
+ * ResumeParserService - Enhanced resume parsing with contact masking
+ * Implements the design specification from design.md
+ */
+class ResumeParserService {
```

Comment:
this class shall handle pdf file. (it is sufficient since pdf is the most common!)